client/src/main/java/io/confluent/kafka/schemaregistry/avro/AvroSchemaUtils.java

```diff
@@ -314,6 +318,58 @@ public static DatumWriter<?> getDatumWriter(
     }
   }
 
+  public static JsonNode findMatchingField(JsonNode node, String path) {
+    String [] identifiers = path.split("\\.");
+    if (identifiers.length < 2) {
+      throw new IllegalArgumentException(
+        String.format("Given field path '%s' has less than two components.", path));
+    }
+
+    // ignore namespace since recordName and fieldName combination is unique in a schema
```

Comment:
I think you can have two records with the same recordName and different namespaces.   You should probably add a test case to verify